repos/system_upgrade/el7toel8/actors/peseventsscanner/libraries/peseventsscanner.py

```diff
@@ -475,44 +486,55 @@ def map_repositories(packages):
+def report_skipped_packages(message, packages, remediation=None):
     """Generate report message about skipped packages"""
     packages = sorted(packages)
     title = 'Packages will not be installed'
     summary = '{} {}\n{}'.format(len(packages), message, '\n'.join(['- ' + p for p in packages]))
-    reporting.create_report([
+    report_content = [
         reporting.Title(title),
         reporting.Summary(summary),
         reporting.Severity(reporting.Severity.HIGH),
         reporting.Tags([reporting.Tags.REPOSITORY]),
-    ] + [reporting.RelatedResource('package', p) for p in packages])
+    ]
+    if remediation:
+        report_content += [reporting.Remediation(hint=remediation)]
+    report_content += [reporting.RelatedResource('package', p) for p in packages]
+    reporting.create_report(report_content)
     if is_verbose():
-        api.show_message(summary)
+        api.current_logger().info(summary)
 
 
 def add_output_pkgs_to_transaction_conf(transaction_configuration, events):
     """
-    Add more packages for removal to transaction configuration if they can be derived as outputs of PES events.
+    Filter out those PES events that conflict with the higher priority transaction configuration files.
 
     Output packages from an event are added to packages for removal only if all input packages are already there.
 
     :param transaction_configuration: RpmTransactionTasks model instance with pkgs to install, keep and remove based
                                       on the user configuration files
     :param events: List of Event tuples, where each event contains event type and input/output pkgs
     """
-    message = 'Marking packages for removal:\n'
+    message = 'The following RHEL 8 packages will not be installed:\n'
 
     for event in events:
         if event.action in (Action.split, Action.merged, Action.replaced, Action.renamed):
             if all([pkg in transaction_configuration.to_remove for pkg in event.in_pkgs]):
                 transaction_configuration.to_remove.extend(event.out_pkgs)
-                message += '- [{action}] {ins} -> {outs}\n'.format(
-                    action=event.action.name,
-                    ins=', '.join(sorted(event.in_pkgs.keys())),
-                    outs=', '.join(sorted(event.out_pkgs.keys()))
+                message += (
+                    '- {outs}\n - Reason: {ins} being {action} {to_or_by} {outs} {is_or_are} mentioned in the'
+                    ' transaction configuration file /etc/leapp/transaction/to_remove\n'.format(
+                        outs=', '.join(event.out_pkgs.keys()),
+                        ins=', '.join(event.in_pkgs.keys()),
+                        action=event.action.name,
+                        to_or_by='by' if event.action == 'Replaced' else 'to',
+                        is_or_are='is' if len(event.in_pkgs.keys()) == 1 else 'are'
+                    )
                 )
 
     api.current_logger().debug(message)
```

Comment:
This message is printed in the log even if there's no pkg to report:
```
2020-07-31 15:31:45.530 DEBUG    PID: 11140 leapp.workflow.FactsCollection.pes_events_scanner: The following RHEL 8 packages will not be installed:

2020-07-31 15:31:45.540 DEBUG    PID: 11140 leapp.workflow.FactsCollection.pes_events_scanner: ---- Processing 36 eligible events for release (7, 5)
```
It's not such a big deal though.

Reply:
ok, will be solved in #549 . Closing this thread